### PR TITLE
Apache Lucene update from 6.1.0 to Apache Lucene 9.5.0 (#82)

### PR DESCRIPTION
**Pull request by BeckerFrank**
I have changed BugzillaAttachmentHandlerTest.java and AllTasksTests.java to get no failures when running on my MacBook with maven.

Please do some tests because I only tried the following cases with eclipse 2022-06 and 2023-03

1. new instance of eclipse and create a query
2. latest released mylyn and  create a query. After this upgrade to the version with Lucene 9.5.0.
(not sure that I have tested all possible cases where we have left some entries with duplicate Date values)

### CODE REVIEW

File: org.eclipse.mylyn-target/mylyn-e4.26.target [hunk 1]
```diff
@@ -134,6 +110,27 @@
 					<version>6.1.0.202203080745-r</version>
 					<type>jar</type>
 				</dependency>
+<!-- https://mvnrepository.com/artifact/org.apache.lucene/lucene-core -->
+<dependency>
+    <groupId>org.apache.lucene</groupId>
+    <artifactId>lucene-core</artifactId>
+    <version>9.5.0</version>
```
Comment (ruspl-afed) [inline org.eclipse.mylyn-target/mylyn-e4.26.target:117]: I wouldn't overtake what the Platform uses, i.e. 9.4.2 https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/master/eclipse.platform.releng.prereqs.sdk/eclipse-sdk-prereqs.target#L32

**Review by ruspl-afed — COMMENTED** [review]
1) It could be wise to not overtake Platform with Lucene version, please see my comment for details
2) We have a number of unrelated changes in .target that looks like formatting. Can we avoid this?

File: org.eclipse.mylyn-target/mylyn-e4.26.target (hunk 1 above)
Comment (ruspl-afed) [reply org.eclipse.mylyn-target/mylyn-e4.26.target:117]: Let me work more on this, it could require both target change and IP review.

Comment (BeckerFrank) [reply org.eclipse.mylyn-target/mylyn-e4.26.target:117]: > Let me work more on this, it could require both target change and IP review.

OK

**Comment by ruspl-afed** [comment]
Progress note:
Waiting for IP review results
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7914
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7915
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7916

**Review by ruspl-afed — APPROVED** [review]
Dependencies are reviewed

**Comment by ruspl-afed** [comment]
> Let me work more on this, it could require both target change and IP review.

It is done @BeckerFrank, please proceed further

**Pull request merged**